webhook/resources/recurringjob/mutator.go

```diff
@@ -52,6 +52,9 @@ func (r *recurringJobMutator) Create(request *admission.Request, newObj runtime.
 	if name != recurringjob.Name {
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/metadata/name", "value": "%s"}`, name))
 	}
+	if recurringjob.Spec.Name == "" {
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/name", "value": "%s"}`, recurringjob.Name))
+	}
```

Comment:
Last question: should we update the `/spec/name` in the upgrade path for the existing jobs?

Reply:
Is there any reason it might be empty during the upgrade?